api_test/src/test/java/ecse429/group7/PriorityStepDefinition.java

```diff
@@ -0,0 +1,130 @@
+package ecse429.group7;
+
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import io.cucumber.java.en.And;
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+
+
+import io.cucumber.datatable.DataTable;
+import kong.unirest.Unirest;
+import kong.unirest.HttpResponse;
+import kong.unirest.JsonNode;
+
+import static org.junit.Assert.*;
+
+import java.util.List;
+import org.junit.runner.RunWith;
+import io.cucumber.junit.Cucumber;
+
+
+public class PriorityStepDefinition extends BaseTest {
+
+    String errorMessage;
+
+    @Given("^the following categories are registered in the todoManagerRestAPI system:$")
+    public void the_following_categories_are_registered_in_the_todomanagerrestapi_system(DataTable table) {
+        List<List<String>> rows = table.asLists(String.class);
+    
+        int idx = 0;
```

Comment:
this `idx` business is interesting, but i think it's a bit confusing. For one thing, idx is not an intuitive name (it really looks like something that would be idiomatic so maybe im not familiar with it), and I also think since it is effectively used as a boolean it could be seamlessly replaced with one. I would do something like `boolean firstRow = false` and set it to true, which would make it super intuitive what's going on, and make the code more readable.